### PR TITLE
Add warning row style

### PR DESCRIPTION
**Pull request by fabiormoura**
### WHY are these changes introduced?

Follows after https://github.com/Shopify/polaris/pull/9872 and a similar motivation as in https://github.com/Shopify/polaris/issues/9873 but for "pending suspension"

image
Storybook url: http://243.27.6.121:6006/

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by fatimasajadi** [comment]
@fabiormoura Could you please provide the storybook link as well?

**Comment by fabiormoura** [comment]
> @fabiormoura Could you please provide the storybook link as well?

http://243.27.6.121:6006/

**Review by fatimasajadi — APPROVED** [review]
LGTM!

File: .changeset/spicy-nails-heal.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': minor
+---
+
+Add a warning status to the IndexTable
```
Comment (chloerice) [inline .changeset/spicy-nails-heal.md]:
```suggestion
Added a `warning` `status` to the `IndexTable.Row`
```

**Review by chloerice — APPROVED** [review]
When you ship this, can you open a PR targeting `next` that cleans up the `#se23` SCSS?

File: .changeset/spicy-nails-heal.md (hunk 1 above)
Comment (fabiormoura) [reply .changeset/spicy-nails-heal.md]: fixed!

**Review by pamelahicks — APPROVED** [review]

**Comment by sophschneider** [comment]
@fabiormoura @pamelahicks I'm merging this PR into our `next` branch in [this PR](https://github.com/Shopify/polaris/pull/10704#issuecomment-1736344080), if you could tophat that would be great!

**Pull request merged**